gma/src/android/gma_android.cc

```diff
@@ -922,18 +923,30 @@ void JNI_NativeAd_completeLoadedAd(JNIEnv* env, jclass clazz, jlong data_ptr,
 
     // Invoke a friend of NativeAdInternal to update its icon image asset.
     GmaInternal::InsertNativeInternalImage(native_ad_internal, icon_internal,
-                                           true, true);
+                                           "icon", true);
     env->DeleteLocalRef(j_icon);
   }
 
+  // getAdChoicesInfo().getImages() can return an empty list and a valid ad can
+  // exist without an adchoices icon image.
+  if (j_adchoices_icon != nullptr) {
+    NativeAdImageInternal adchoices_icon_internal;
+    adchoices_icon_internal.native_ad_image = j_adchoices_icon;
```

Comment:
Since j_adchoices_icon is passed into this callback, it's probably a local reference - does this need to become a global reference instead?

Reply:
While we perform InsertNativeInternalImage below on line 937, it creates a GlobalRef for j_adchoices_icon.
ref: https://github.com/firebase/firebase-cpp-sdk/blob/main/gma/src/android/native_ad_image_android.cc#L61

So, i believe we can have j_adchoices_icon as a local reference here and safely delete it, after a succesful InsertNativeInternalImage.

Reply:
OK, sounds good.